src/main/java/org/terasology/las/LaSExtraPositionFilters.java

```diff
@@ -0,0 +1,77 @@
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.las;
+
+import com.google.common.base.Predicate;
+import org.joml.Vector3i;
+import org.terasology.engine.world.generation.facets.SurfacesFacet;
+import org.terasology.ligthandshadow.componentsystem.LASUtils;
+
+
+
+/**
+ * A collection of filters that restrict the placement of objects, specific to Light and Shadow.
+ *
+ */
+
+public final class LaSExtraPositionFilters {
+
+    private LaSExtraPositionFilters() {
+        // no instances
+    }
+
+    /**
+     * Filters based on surface flatness
+     *
+     * @param surfaceFacet the surface facet that contains all tested coords.
+     * @return a predicate that returns true only if the facet would not overlap the red or black base.
+     */
+    public static Predicate<Vector3i> notOnBase(final SurfacesFacet surfaceFacet) {
+
+        return new Predicate<Vector3i>() {
+
+            @Override
+            public boolean apply(Vector3i input) {
+                int myx = input.x();
+                int myz = input.z();
+                int blackx = LASUtils.CENTER_BLACK_BASE_POSITION.x();
+                int blackz = LASUtils.CENTER_BLACK_BASE_POSITION.z();
+                int redx = LASUtils.CENTER_RED_BASE_POSITION.x();
+                int redz = LASUtils.CENTER_RED_BASE_POSITION.z();
+
+                for (int xOffset = -1*(LASUtils.BASE_EXTENT); xOffset <= LASUtils.BASE_EXTENT; xOffset++) {
+                    for (int zOffset = -1*(LASUtils.BASE_EXTENT); zOffset <= LASUtils.BASE_EXTENT; zOffset++) {
+                        if(     (myx == blackx + xOffset && myz == blackz + zOffset)||
+                                (myx == redx + xOffset && myz == redz + zOffset)||
+                                (myx-1 == blackx + xOffset && myz == blackz + zOffset)||
+                                (myx-1 == redx + xOffset && myz == redz + zOffset)||
+                                (myx+1 == blackx + xOffset && myz == blackz + zOffset)||
+                                (myx+1 == redx + xOffset && myz == redz + zOffset)||
+                                (myx == blackx + xOffset && myz-1 == blackz + zOffset)||
+                                (myx == redx + xOffset && myz-1 == redz + zOffset)||
+                                (myx == blackx + xOffset && myz+1 == blackz + zOffset)||
+                                (myx == redx + xOffset && myz+1 == redz + zOffset))
+
+                            return false;
+                    }
+                }
+                return true;
+            }
+        };
```

Comment:
There isn't really a need to have this separated out from your implementation. stuff like this is better to do after the fact when you realize that it would make sense to separate it out as a a utility.

Reply:
> 
> 
> There isn't really a need to have this separated out from your implementation. Better to do this type of separate after the fact when you are using the same logic throughout the project. Makes it hard to understand what is happening when the details are separated out into another utility class.

Hello, and thanks very much for your response and review.

There may be a benefit to adding more position filters, specific to Light&Shadow, rather than declaring them anonymously in the Tree Provider. Another issue I'm working on resolving is that the tallest trees penetrate the Floating Platform when they spawn under it (this could be solved by restricting trees from spawning there, but it may be simpler and more effective to raise the height of the floating platform).

This version of LaSTreeProvider uses copied class code from DefaultTreeProvider which I edited to add the new filter. I would also ask that you look at [my previous commit ](https://github.com/Terasology/LightAndShadow/pull/198/commits/8a4410394eef6274f86854d086a271a4dcc31b09) in which LaSTreeProvider inherits from DefaultTreeProvider and overrides the getFilters() method. While this is much more readable, the game would not build. Talking to other developers on Discord, it was suggested that this may have been due to a permission/whitelist issue that did not come up when I copied the class code.

What do you think?